### DISCUSSION TITLE
Is everybody OK if I were to transfer everything to pugjs?

### DISCUSSION THREAD

**Discussion by firefish111** [Repl-it-Coders/Repl.it-2020#3] [Ideas]
###### read the title
Another thing is that the most upvoted posts are in concrete, and not dynamically fetched.
Also, the CSS really hurts my eyes. How is a css stylesheet similar to [this](https://weekly-challenges-the-comeback.firefish.repl.co/chal)?
It doesn't give you a headache.
Please comment your (dis)approvals. Thank you!

**Comment by firefish111** [comment]
@whippingdot @Repl-it-Coders/nodejs @Repl-it-Coders/web-dev

**Reply by cursorweb** [reply]
no thanks

**Comment by AmazingMech2418** [comment]
@firefish111 
PugJS? \*cringe\*

1. Unneeded dependencies
2. Lack of control
3. I don't know it
4. EJS is better anyways

**Reply by firefish111** [reply]
1. Space is cheap
2. As the old saying goes, `When there's a will, there's a way`
3. Learn it
4. That's bias

**Reply by AmazingMech2418** [reply]
@firefish111 
> 1. Space is cheap
> 2. As the old saying goes, `When there's a will, there's a way`
> 3. Learn it
> 4. That's bias

1. Until you run out of storage in Repl.it... Dependencies = :-1:
2. Yeah, but it's easier in HTML or EJS or something.
3. Why would I want to learn something that bad? LOL!
4. It's truth! LOL!

**Reply by firefish111** [reply]
1. haha hacker plan go brrr
2. EJS is ugly
3. ***See number 4***
4. I find those remarks insulting, let me tell you a true story.
When me and HahaYes were working on HahaChat, Coder100 insisted on converting the html to pugjs.
Me and HahaYes were clueless as to what pugjs was, and just thought he deliberately borked the current code, and just ran away.
However, when we asked him what he was doing, he said he was in the middle of something.
What happened was that he left in the middle of moving `index.html` up a directory, and translating it into `pugjs`. But as we had no idea what `pugjs` was, we started to question what his motives were.
As I had coined the phrase `grilling apricots on the barbecue` at the time, HahaYes made a parody of it in response to Coder100's actions. To be precise, se said `grilling cookies on a stake` as Coder100's pfp is a cookie. This brought him the nickname `grilled cookie`.
Another thing he did was change the listening port from `6111` to `8080` thus also earning him the nickname `Port 8080`. Which of course, was a phrase to good to not be entered into the Urban Dictionary.
For some time afterwards, I hated Coder100, `express`, and `pugjs`. A month later, I was developing the parody game `cookie-griller` about grilling cookies. I wanted to implement repl.it auth, and soon afterward, discovered the only way to do this was to use express. Coder100 used `pugjs` in his tutorial, and I decided to give it a try. So converting my spaghetti http code, into clean express code, deserved some adventuring into `pugjs`. I tried it, and liked it ever since. Later, I took one single glance at some `ejs` code and immediately knew that it was extremely ugly. I mean, the `%` by itself, is ugly. Pairing it with an angle bracket (`<%`), is even more ugly. 
Even though `pugjs` is `YAML`-like, and relies heavily on indentation, it is still far superior to old-fashioned `ejs`.

# Is this long? You will not regret reading it.

**Reply by ghost** [reply]
You will also not regret buying our product for 3 pounds 99.
###### Bang, and the dirt is gone!

**Reply by whippingdot** [reply]
Laughs in American

**Reply by ghost** [reply]
@whippingdot and you could save money with our -100% discount.

[Buy Now!](https://fwesh.yonle.repl.co)

###### This is not a real product, only reference to other cleaning based solutions and gags. You agree that by not reading the fine print super fast, you are liable for your own actions in whatever you do. You cannot buy this product in any region. We love capitalism and the law.

**Reply by ghost** [reply]
@whippingdot don't click the link.

**Reply by whippingdot** [reply]
I didn't

why?

**Reply by programmeruser2** [reply]
rickroll

**Reply by whippingdot** [reply]
oh lol

**Reply by whippingdot** [reply]
@programmeruser2 are you still on replit? I feel like most of the ogs stopped posting. Cookie and bookie post once in a while, fuzzy died, tussiez used to post but stopped getting traction...Rayhan stopped posting, Tsunami died, I remember some other big dude who started with an r I think - he also stopped posting, codemonkey like never posts, csharp posts once in a blue moon, squid is a mod but I barely see him active.

I am guessing you also stopped posting right? What are your current projects then. Do you still code?

Lots of questions cause I haven't really ever talked to you.

**Reply by ghost** [reply]
@whippingdot cough cough. He was head of the project that you are now on. He just gave me the role and so I just went with it.

**Reply by programmeruser2** [reply]
> @programmeruser2 are you still on replit? I feel like most of the ogs stopped posting. Cookie and bookie post once in a while, fuzzy died, tussiez used to post but stopped getting traction...Rayhan stopped posting, Tsunami died, I remember some other big dude who started with an r I think - he also stopped posting, codemonkey like never posts, csharp posts once in a blue moon, squid is a mod but I barely see him active.
> 
> I am guessing you also stopped posting right? What are your current projects then. Do you still code?
> 
> Lots of questions cause I haven't really ever talked to you.

ask me on discord

**Reply by programmeruser2** [reply]
> @whippingdot cough cough. He was head of the project that you are now on. He just gave me the role and so I just went with it.

what project?

**Reply by firefish111** [reply]
> @programmeruser2 are you still on replit? I feel like most of the ogs stopped posting. Cookie and bookie post once in a while, fuzzy died, tussiez used to post but stopped getting traction...Rayhan stopped posting, Tsunami died, I remember some other big dude who started with an r I think - he also stopped posting, codemonkey like never posts, csharp posts once in a blue moon, squid is a mod but I barely see him active.
> 
> I am guessing you also stopped posting right? What are your current projects then. Do you still code?
> 
> Lots of questions cause I haven't really ever talked to you.

@whippingdot there's a very good reason why no-one uses replit anymore: the service has become dogshit. UI revamp after UI revamp,  there's no reason to stay.

**Reply by programmeruser2** [reply]
> > @programmeruser2 are you still on replit? I feel like most of the ogs stopped posting. Cookie and bookie post once in a while, fuzzy died, tussiez used to post but stopped getting traction...Rayhan stopped posting, Tsunami died, I remember some other big dude who started with an r I think - he also stopped posting, codemonkey like never posts, csharp posts once in a blue moon, squid is a mod but I barely see him active.
> > I am guessing you also stopped posting right? What are your current projects then. Do you still code?
> > Lots of questions cause I haven't really ever talked to you.
> 
> @whippingdot there's a very good reason why no-one uses replit anymore: the service has become dogshit. UI revamp after UI revamp, there's no reason to stay.

true, and I stopped programming

**Reply by whippingdot** [reply]
> > @programmeruser2 are you still on replit? I feel like most of the ogs stopped posting. Cookie and bookie post once in a while, fuzzy died, tussiez used to post but stopped getting traction...Rayhan stopped posting, Tsunami died, I remember some other big dude who started with an r I think - he also stopped posting, codemonkey like never posts, csharp posts once in a blue moon, squid is a mod but I barely see him active.
> 
> > 
> 
> > I am guessing you also stopped posting right? What are your current projects then. Do you still code?
> 
> > 
> 
> > Lots of questions cause I haven't really ever talked to you.
> 
> 
> 
> ask me on discord

I'll add u

You are in the server right?

**Reply by whippingdot** [reply]
> > > @programmeruser2 are you still on replit? I feel like most of the ogs stopped posting. Cookie and bookie post once in a while, fuzzy died, tussiez used to post but stopped getting traction...Rayhan stopped posting, Tsunami died, I remember some other big dude who started with an r I think - he also stopped posting, codemonkey like never posts, csharp posts once in a blue moon, squid is a mod but I barely see him active.
> 
> > > I am guessing you also stopped posting right? What are your current projects then. Do you still code?
> 
> > > Lots of questions cause I haven't really ever talked to you.
> 
> > 
> 
> > @whippingdot there's a very good reason why no-one uses replit anymore: the service has become dogshit. UI revamp after UI revamp, there's no reason to stay.
> 
> 
> 
> true, and I stopped programming

Woah stopped coding

**Reply by programmeruser2** [reply]
> > > @programmeruser2 are you still on replit? I feel like most of the ogs stopped posting. Cookie and bookie post once in a while, fuzzy died, tussiez used to post but stopped getting traction...Rayhan stopped posting, Tsunami died, I remember some other big dude who started with an r I think - he also stopped posting, codemonkey like never posts, csharp posts once in a blue moon, squid is a mod but I barely see him active.
> > 
> > 
> > > 
> > 
> > 
> > > I am guessing you also stopped posting right? What are your current projects then. Do you still code?
> > 
> > 
> > > 
> > 
> > 
> > > Lots of questions cause I haven't really ever talked to you.
> > 
> > 
> > ask me on discord
> 
> I'll add u
> 
> You are in the server right?

yes

**Reply by firefish111** [reply]
> > > @programmeruser2 are you still on replit? I feel like most of the ogs stopped posting. Cookie and bookie post once in a while, fuzzy died, tussiez used to post but stopped getting traction...Rayhan stopped posting, Tsunami died, I remember some other big dude who started with an r I think - he also stopped posting, codemonkey like never posts, csharp posts once in a blue moon, squid is a mod but I barely see him active.
> > > I am guessing you also stopped posting right? What are your current projects then. Do you still code?
> > > Lots of questions cause I haven't really ever talked to you.
> > 
> > 
> > @whippingdot there's a very good reason why no-one uses replit anymore: the service has become dogshit. UI revamp after UI revamp, there's no reason to stay.
> 
> true, and I stopped programming

@programmeruser2 wtf why stop programming

**Reply by programmeruser2** [reply]
idk
i started again

**Comment by AmazingMech2418** [comment]
@firefish111 

1. Hacker plan doesn't give *unlimited* storage...
2. Not sure if you heard of it, but there's this thing called *CSS*. And if you're talking about syntax, yeah. Pure HTML is better. LOL! Or a self-made templating engine...
3. (Ah! Markdown, why no custom numbering?) Yeah, port 8080 is better, but dependencies are trash. LOL! Instead of `express`, use `serverfire` (1/7 of the size of `express` with similar syntax and the same functionality) or something and instead of `pugjs`, either make your own templating engine or just use HTML. LOL!

**Reply by firefish111** [reply]
1. Pretty close
2. Well, at last, something we agree on.
3. Stop being obsessed with size! I tried to make my own template engine, as referred to in `spaghetti http code`.

**Comment by programmeruser2** [comment]
Pugjs has unnecessary dependencies, and EJS should be readable to anyone how has knowledge of HTML and JS.

**Reply by firefish111** [reply]
Google is on my side here.
![image](https://user-images.githubusercontent.com/33234914/103490920-07fa7180-4e17-11eb-851a-158985675ca5.png)

**Reply by programmeruser2** [reply]
Yes, Pug's syntax can be more readable. Perhaps we could try Handlebars?

**Reply by firefish111** [reply]
Well, Handlebars is ~~suspiciously a lot like vuejs~~ quite interesting. It doesn't seem to have any middleware for express however.

**Reply by programmeruser2** [reply]
It does have a middleware for Express. See https://www.npmjs.com/package/express-handlebars.

**Reply by firefish111** [reply]
> A Handlebars view engine for Express ***which doesn't suck.***

Ok I'm now extremely interested, i'll do some testing.

**Reply by cursorweb** [reply]
handlebars is literally not vuejs lmao

**Reply by firefish111** [reply]
but the syntax, it uses double brace @cursorweb

**Reply by programmeruser2** [reply]
> Pug's syntax ~~can be~~ is more readable
> EJS should**n't** be readable

**Reply by programmeruser2** [reply]
i changed my mind as you can see

**Reply by whippingdot** [reply]
why did you die for a year

I tried to ping you several times but you never responded @programmeruser2

**Reply by programmeruser2** [reply]
hmm where
i only clicked on the notifs to get rid of them

**Comment by tsunami-or-something** [comment]
where do i turn notifs off

**Reply by firefish111** [reply]
no clue, sorry :/

**Reply by whippingdot** [reply]
@tsunami-or-something you go to settings, you go to notifications, and then you untick everything.

**Comment by whippingdot** [comment]
absolutely not, also I think the CSS is nice. The one that is not nice is the repl auth CSS. I personally like the CSS of the website. Your website's CSS is awesome too but we can't make everything in the same theme.

**Comment by rohilpatel1** [comment]
Lmao yeah
Sincerely,
Rohil Patel
On Jan 3, 2021, at 5:03 PM, tsunami <notifications@github.com> wrote:
﻿
where do i turn notifs off
—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly,<https://github.com/Repl-it-Coders/Repl.it-2020/discussions/3#discussioncomment-259050> view it on GitHub, or<https://github.com/notifications/unsubscribe-auth/ALUEPMZB2AHXG6IBG5KZXTDSYDZUPANCNFSM4VSD5GYA> unsubscribe.
Notice
This E-mail transmission may contain confidential or legally privileged information that is intended for the individual or entity named in the E-mail address. Use of such information by any intended recipient shall be limited to the purpose for which such information was sent. Unauthorized use, disclosure, or copying is strictly prohibited. If you received this E-mail transmission in error, please reply to the sender and delete the message. Thank you.

**Reply by whippingdot** [reply]
what???? Also, why are you replying through email, it is hard to read.

**Comment by Isaiah-Day** [comment]
Dude. I take ONE 10 hour car ride and DO YOU HAVE ANY IDEA HOW MANY NOTIFICATIONS I HAVE!!! (But at least there was a good story.)

**Comment by Isaiah-Day** [comment]
Also, i feel fine about switching to `pug.js`.

**Comment by ghost** [comment]
The issue with PUG.JS is that it changes the html sytax, we would be better off using vue or react than pug as it offers the same things.

**Reply by ghost** [reply]
Scratch that,  I just use Vue.JS now. I do not like react.

**Reply by whippingdot** [reply]
I don't like either lol. But then again I'm not a web developer

**Comment by ghost** [comment]
I am resolving that with replit - performance feedback was given alongside my experience with the mobile version. I might end up making my own IDE at this rate though after my team's app comes out.
________________________________

**Comment by ghost** [comment]
Yeah, I did that as well for a little bit before - but that was because I burnt myself out which is my fault.

**Reply by whippingdot** [reply]
?
